src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -474,18 +474,19 @@ def validate_metadata(
     interpreter_constraints: InterpreterConstraints,
     lockfile: Lockfile,
     consumed_req_strings: Iterable[str],
+    validate_consumed_req_strings: bool,
```

Comment:
We now always pass the req strings through so we can use them to give more context to the user in the error message, so we add this flag to determine whether we should validate them (previously we did this by not passing the req strings in)